tests/src/ut_test_kvp.c

```diff
@@ -434,6 +568,78 @@ static int test_ut_kvp_createGlobalJSONInstance( void )
     return 0;
 }
 
+static int test_ut_kvp_createGlobalYAMLInstanceForMallocedData( void )
+{
+    ut_kvp_status_t status;
+    test_ut_memory_t kvpMemory;
+
+    gpMainTestInstance = ut_kvp_createInstance();
+    if ( gpMainTestInstance == NULL )
+    {
+        assert( gpMainTestInstance != NULL );
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return -1;
+    }
+
+    if (read_file_into_memory(KVP_VALID_TEST_YAML_FILE, &kvpMemory) == 0)
+    {
+        status = ut_kvp_openMemory(gpMainTestInstance, kvpMemory.buffer, kvpMemory.length);
+        UT_ASSERT(status == UT_KVP_STATUS_SUCCESS);
+        status = status;
+    }
+
+    if ( status != UT_KVP_STATUS_SUCCESS )
+    {
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return -1;
+    }
+
+    return 0;
+}
+
+static int test_ut_kvp_createGlobalJSONInstanceForMallocedData( void )
+{
+    ut_kvp_status_t status;
+    test_ut_memory_t kvpMemory;
+
+    gpMainTestInstance = ut_kvp_createInstance();
+    if ( gpMainTestInstance == NULL )
+    {
+        assert( gpMainTestInstance != NULL );
+        UT_LOG_ERROR("ut_kvp_open() - Read Failure");
+        return -1;
+    }
+
+    if (read_file_into_memory(KVP_VALID_TEST_JSON_FILE, &kvpMemory) == 0)
```

Comment:
this memory must be freed during the shutdown function, therefore it must be &gkvpMemory

Reply:
file moved to ut-control